### ISSUE TITLE
Feature request: Wait for successful job start

### ISSUE THREAD

**Issue by james-masson** [terraform-providers/terraform-provider-nomad#56]
```
Terraform v0.11.8
+ provider.nomad v1.3.0
```

One thing I've come to expect from Terraform is that it waits for success on the infrastructure object it creates, as much as is possible.

eg. Terraform waits on creating a AWS Instance, until the instance is running - not just that the AWS API has given a 200 OK back on submitting the HTTP request.

When creating new jobs with the nomad provider it seems like all it looks for is that the API accepted the request - not that the job could be started.

I would like to be able to wait for the job to start, to enable the following use cases.

1. Using the properties of the job that are only available while it's active in downstream objects.
2. Using batch tasks as a form of lifecycle hooks ( before changing service A, run batch B) - as nomad itself seems to be missing this concept, and terraform is a good place for this dependency-graph driven execution.

**Comment by sparkacus** [comment]
Is there a way to have the detach option support the new lifecycle configuration? 0https://www.nomadproject.io/docs/job-specification/lifecycle/

I'd like a way to only mark a deployment as success if all tasks are started successfully and without error.

**Comment by cgbaker** [comment]
@sparkacus , do you mind opening a new issue, with some specifics about your configuration, what behavior you are observing, and what you'd like it to be. thanks!
